### PR TITLE
Master cleaning

### PR DESCRIPTION
**Pull request by zakkak**
Final step towards Proposal 1 of https://oss.oracle.com/pipermail/graalvm-dev/2020-July/000071.html:

> Make the default branch contain only a README file explaining what Mandrel is, how it relates to Graal, and how to get your hands on it. This branch will also contain any workflows related to housekeeping, syncing branches, nightly testing, etc.
> This branch will not contain any code. So after git clone https://github.com/graalvm/mandrel and reading the README users will be asked to checkout another branch if they want to access the source code.

This PR removes all code from `master` and updates the readme and contributing files accordingly.
After approval I propose purging the `master` branch history (this will make `git clone` fast). 

PS: now is a good time to think about using a different (than `master`) name for the default branch, please see #102 .

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
-and academia so far and we thank you for considering to contribute your changes!
+# Contributing to Mandrel
 
-- Learn [how to become a GraalVM contributor](https://www.graalvm.org/community/contributors/).
-- Subscribe and post to [graalvm-dev@oss.oracle.com](https://oss.oracle.com/mailman/listinfo/graalvm-dev) for questions related to working with the sources or extending the GraalVM ecosystem by creating new languages, tools, or embeddings.
+Mandrel is a downstream distribution of [GraalVM CE](www.github.com/oracle/graal).
+As a result contributions to Mandrel should be limited to:
+
+1. Backports from the upstream repository of GraalVM (e.g. bringing a feature/bug-fix from GraalVM CE 20.2 to Mandrel 20.1)
+2. Mandrel specific bug fixes (e.g. patches to make it work with upstream OpenJDK instead of Oracle's labs OpenJDK)
```
Comment (jerboaa) [inline CONTRIBUTING.md:7]: Suggested phrasing: "... instead of Oracle Labs JDK"

Comment (zakkak) [reply CONTRIBUTING.md:7]: :+1: I will use `LabsJDK` as mentioned in https://github.com/graalvm/labs-openjdk-11/blob/master/README.md

**Comment by zakkak** [comment]
> OK. I don't think we should change the default branch. It's the one showing the readme on Githubs landing page when visiting graalvm/mandrel. The changed readme on master after this PR will have all the info one needs. Thoughts?

Oh, sorry, I didn't make it clear. By changing the default branch I was referring to changing it's name from `master` to something else, e.g. `main`, `readme`, `welcome`, etc. (I edited the 1st comment to make it clear)

**Comment by jerboaa** [comment]
> > OK. I don't think we should change the default branch. It's the one showing the readme on Githubs landing page when visiting graalvm/mandrel. The changed readme on master after this PR will have all the info one needs. Thoughts?
> 
> Oh, sorry, I didn't make it clear. By changing the default branch I was referring to changing it's name from `master` to something else, e.g. `main`, `readme`, `welcome`, etc. (I edited the 1st comment to make it clear)

OK. It's not clear why we'd need to rename it, though.

**Comment by zakkak** [comment]
> OK. It's not clear why we'd need to rename it, though.

Just to avoid the use of the word "master" in an effort to become more inclusive (following the example of others).

**Comment by jerboaa** [comment]
> > OK. It's not clear why we'd need to rename it, though.
> 
> Just to avoid the use of the word "master" in an effort to become more inclusive (following the example of others).

OK, that makes sense.

**Comment by maxandersen** [comment]
having different default names for branches is annoying. I suggest we raise the request to change it on graalvm org level rather than only changing it in one part of the github organization.

**Comment by zakkak** [comment]
> having different default names for branches is annoying. 

Yes I can see that happening, but given that in Mandrel we are **not** developing on `master` I don't think it's a big issue for us.

> I suggest we raise the request to change it on graalvm org level rather than only changing it in one part of the github organization.

TBH I don't see how that helps. Mandrel (being a downstream project) is already working in a totally different way than the rest of the projects under the graalvm org and the inconvenience if any would be minimal for its users. On the contrary, asking all other projects under graalvm org to change is probably too much to ask and would cause much more inconvenience to their users.

In any case I consider this a separate issue, so if there is no consensus I would prefer to discuss it on it's own issue.
@maxandersen could you please approve this PR if it looks OK to you (the PR doesn't include the renaming)?

**Comment by zakkak** [comment]
I created #102 for the default naming discussion.

**Comment by zakkak** [comment]
@adinn @maxandersen I would like at least one more approval from one of you two before merging

**Comment by zakkak** [comment]
Merged and force pushed to master with clean history

**Pull request merged**